### PR TITLE
Luis copy button weekly summary

### PR DESCRIPTION
**Pull request by luisarevalo21**
# Description
Screenshot 2023-08-22 at 9 16 14 PM

## Main changes explained:

Added a copy icon to user's profile component. When viewing weekly summaries admins can quickly copy and paste their team members summaries to speed up their managerial tasks. 
Jae also suggested to add the icon to the weekly reports summary to view everyone's weekly summaries to counter-act the current bug. This way admins can copy weekly summaries there as well. 

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user or owner.
4. go to dashboard→ view profile → show weekly summary → select a team member.
5. View their weekly summaries and click on the copy button to copy their summary to your clipboard. 
Screenshot 2023-08-22 at 9 23 55 PM

7. Verify clipboard has the exact text copied to your clipboard. 
Screenshot 2023-08-22 at 9 11 15 PM

8. Repeat from steps 3  but navigate to  dashboard→ reports → weekly summary reports  → select a team member.
9. View their weekly summaries and click on the copy button to copy their summary to your clipboard. 
Screenshot 2023-08-22 at 9 15 21 PM

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/22454953/b13ec2c9-40a5-4fa5-b24d-c0611bb02ec3

### CODE REVIEW

File: src/components/UserProfile/TeamWeeklySummaries/TeamWeeklySummaries.css [hunk 1]
```diff
@@ -27,3 +31,24 @@
...
+  top: 0;
+  margin: 2px 2px; */
+
+.copy-icon {
+  margin: 2px 2px;
+}
+.team-weekly-summary-container {
+  display: flex;
+  flex-direction: row;
+  align-items: start;
```
Comment (yurisokolovicz) [inline src/components/UserProfile/TeamWeeklySummaries/TeamWeeklySummaries.css:48]: Change for:

align-items: flex-start;

**Review by yurisokolovicz — CHANGES_REQUESTED** [review]
Dear @luisarevalo21 

The error that appeared in my console is a warning related to the CSS stylesheet file "TeamWeeklySummaries.css". The recommendation is to consider replacing the value "start" with "flex-start." This is advised because "flex-start" is the default value for aligning content within a flex container.

**Before adding flex-start:**


Screenshot from 2023-08-23 11-23-34


**After adding flex-start:**

Screenshot from 2023-08-23 11-29-11

**Review by harshidad — COMMENTED** [review]
1. dashboard→ view profile → show weekly summary → select a team member.
      View their weekly summaries and click on the copy button to copy their summary to your clipboard.

2. navigate to dashboard→ reports → weekly summary reports → select a team member.
      View their weekly summaries and click on the copy button to copy their summary to your clipboard.

In both the scenario mentioned above, copy button working fine. Only I observed that, at the end "&nbsp" coming, even though the space, I gave in each sentence. But still in last sente
Screenshot -1_button weekly summary #1166
Screenshot -2 _button weekly summary #1166
Screenshot -3_button weekly summary #1166
nce additional text display &nbsp. 

Please check the attched screenshot.

**Review by iTvX — CHANGES_REQUESTED** [review]
I checked you pr, copy paste looks fine.


The title bar does not display all the buttons. I have tried the development branch, and this issue is not present.
CleanShot 2023-08-23 at 13 44 05@2x

**Review by DavidC0126 — COMMENTED** [review]
Hi @luisarevalo21 I tested your PR and it works as intended, just the problem of other links not visible as an ADMIN user (not related to this PR).
copy_report

File: src/components/UserProfile/TeamWeeklySummaries/TeamWeeklySummaries.css (hunk 1 above)
Comment (luisarevalo21) [reply src/components/UserProfile/TeamWeeklySummaries/TeamWeeklySummaries.css:48]: Hello @yurisokolovicz
I've fixed the code with your suggestion. 
Thanks!

**Comment by luisarevalo21** [comment]
> I checked your pr, copy paste looks fine. 
> 
> The title bar does not display all the buttons. I have tried the development branch, and this issue is not present. CleanShot 2023-08-23 at 13 44 05@2x

Hello @iTvX 

What type account are you using? 
As when I login as an admin or an owner, I can view everything.
If you are logging into as a volunteer, volunteers do not have access to the "suggest changes on a task" 
I logged in via the development branch and saw this. 

See screenshot below. 
Screenshot 2023-08-23 at 7 48 38 PM
 

Screenshot 2023-08-23 at 7 44 58 PM

**Review by Haoj1 — CHANGES_REQUESTED** [review]
Copy the summaries in profile seems to be good. However, there is a memory leak error when I trying to open weekly summary reports page and a non reference error by FontAwesome. See the screenshots.


Screen Shot 2023-08-24 at 12 03 33 AM


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/99279909/51338bcf-7165-4027-a90a-fa2f41893f54

**Review by veronicacheng2 — COMMENTED** [review]
Hi @luisarevalo21, I have tried testing the PR. Unfortunately, I can only test the copy function from the user profile because currently the other links have disappeared.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/95119686/65971e6a-fd1e-4c29-ae95-7a968dcfa8a1

**Review by belagaam — DISMISSED** [review]
Tested the PR as per the instructions. Everything works as expected.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/47104198/e2217adb-14b3-4179-8260-ac9b035a9973

**Comment by Jiyuanxia** [comment]
Hi @luisarevalo21 , I have reviewed your pull request and tested it. Copy button works as you intended but I met some problems when I open weekly summaries report page.
Screenshot 2023-08-24 171357
Screenshot 2023-08-24 172019

**Comment by tsunami776** [comment]
Tested with Admin and Owner account. The copy functionality works well on the user's profile. I can see the 'Weekly Summaries Report' button, however, the page crashes when I click on the button, and the browser console gives a memory leak error. 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/2bb5d3a5-9bc3-485d-a0b2-741fcfe0f1fe

File: src/components/UserProfile/TeamWeeklySummaries/TeamWeeklySummaries.css (hunk 1 above)
Comment (yurisokolovicz) [reply src/components/UserProfile/TeamWeeklySummaries/TeamWeeklySummaries.css]: Thank you very much!

**Comment by luisarevalo21** [comment]
> Copy the summaries in profile seems to be good. However, there is a memory leak error when I trying to open weekly summary reports page and a non reference error by FontAwesome. See the screenshots.
> 
>  Screen Shot 2023-08-24 at 12 03 33 AM
>  1182.mov

Fixed! Please try testing again. Thanks

**Comment by luisarevalo21** [comment]
> Tested with Admin and Owner account. The copy functionality works well on the user's profile. I can see the 'Weekly Summaries Report' button, however, the page crashes when I click on the button, and the browser console gives a memory leak error.
> 
>  PR.Review.1166.mp4

Fixed! Please try testing again. Thanks

**Comment by luisarevalo21** [comment]
> Hi @luisarevalo21 , I have reviewed your pull request and tested it. Copy button works as you intended but I met some problems when I open weekly summaries report page. Screenshot 2023-08-24 171357 Screenshot 2023-08-24 172019

Hello, I adjusted the code and fixed it. Please retest it again. Thanks

**Comment by luisarevalo21** [comment]
> Hi @luisarevalo21, I have tried testing the PR. Unfortunately, I can only test the copy function from the user profile because currently the other links have disappeared.
> 
>  HGN.APP.-.Google.Chrome.2023-08-23.23-39-56.mp4

Hello @veronicacheng2  
This was an ongoing bug that occured on Monday. Please switch to another branch that was posted before Monday and try switching to this branch after. 
If that doesn't work. I had to restore my entire development env and now it works. Feel free to reach out. 
Thanks.

**Comment by luisarevalo21** [comment]
> Copy the summaries in profile seems to be good. However, there is a memory leak error when I trying to open weekly summary reports page and a non reference error by FontAwesome. See the screenshots.
> 
>  Screen Shot 2023-08-24 at 12 03 33 AM
>  1182.mov

Hello ! I fixed the issue that occurred in FormattedReportsJSX
As for the memory leak it occurs sometimes but is unrelated to this PR.

Please retest the code to make Formmated Reports does not crash.
Thanks

**Comment by n0vay** [comment]
Unable to test dues to missing dashboard icons error.

**Review by liaoyufu-leo — APPROVED** [review]
Hi the copy button shows correct on my device! @luisarevalo21 
image
image

**Comment by luisarevalo21** [comment]
> Unable to test dues to missing dashboard icons error.

Hello can you elaborate a bit more on the error?
Are you referencing to the the missing links on the navbar?

Thanks

**Comment by arcan9** [comment]
> If that doesn't work. I had to restore my entire development env and now it works. Feel free to reach out.
> Thanks.

HI @luisarevalo21. I'm also trying to test this but unable to due to the missing navbar links. By restoring the entire development environment, do you mean just re-cloning the HighestGoodNetwork repo?

**Review by yaow62023 — DISMISSED** [review]
I tested your pull request, and the copy text function works properly under both the "View Profile" and "Weekly Summary Report" options. The copy buttons are also functioning correctly. However, I also have a question: I noticed that there are "&nbsp;" occurrences for spaces and line breaks. Is there a way to prevent this from happening? 


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/132e4a08-51d4-497c-8f3b-25604b96256a

**Review by yurisokolovicz — APPROVED** [review]
Thank you!

**Review by tsunami776 — DISMISSED** [review]
The Weekly Summaries Report page loads correctly without crashing. The copy function works on both the user's profile and the Weekly Summaries Report.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/43768723/9a5802b1-4a79-4a1b-aa89-8bea087d8117

**Review by Indujaks — APPROVED** [review]
Tested the changes, copy function works perfectly, Good Job!

Screenshot 2023-08-30 at 4 03 59 PM

Screenshot 2023-08-30 at 4 05 16 PM

**Comment by beblicarl** [comment]
The weekly summary report page took so long to load however this functionality works as intended

[screencast-bpconcjcammlapcogcnnelfmaeghhagj-2023.09.04-11_00_47.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/2da3c9ee-2719-46f1-81fd-7e623218054c)

**Review by beblicarl — APPROVED** [review]

**Comment by one-community** [comment]
Thank you all! Resolve merge conflicts and I'll merge this one!

**Review by GaryB93 — CHANGES_REQUESTED** [review]
Is this branch still under review? I tested the functionality of the copy button and it works as intended. I did notice on the Weekly Summaries Report page when you copy a summary it adds "&nbsp" to the end of a line. Could be that specific summary, as when I tested the button on my own summary on the user profile page I didn't get any "&nbsp".

Also I have some concerns with some of the file changes starting with some files being added back in to the eslintignore and prettierignore files. Seems like this branch has had a lot of changes and just want to make sure it has the most up-to-date development branch changes.

Screenshot 2023-09-07 at 2 39 14 PM
Screenshot 2023-09-07 at 3 19 40 PM

**Comment by Shubhu0** [comment]
Hi @luisarevalo21 I tested and reviewed your PR. It worked perfectly and displayed the summary as well as presented the copy button for the summary. Another thing to note would be that the page had a huge buffer period and took a while to load.

image

**Review by JeffLi117 — CHANGES_REQUESTED** [review]
### Results using profile method

Report to be copied:
image

Toast once copied:
image

Text snippet pasted into Word doc:
image

### Results using reports method

Report to be copied:
image

Toast once copied:
image

Text snippet pasted into Word doc:
image
(*Note, I also get the exact same output copying & pasting through the view profile pathway - I just chose a different summary initially)

### Changes requested
Using both methods you described in the PR, you can see that copying text that has a whitespace at the end of the line leads to pasting a "&nbsp;" at the end - this will probably slow down the work flow, which will be detrimental to the purpose of this PR. 

Adding console log within the (hopefully correct) component:
image

The console:
image

This may be outside the scope of the files that you changed. It could be the way the data is being presented/parsed before copying. Nevertheless, it seems to be an issue worth addressing before merging.

**Review by JeffLi117 — DISMISSED** [review]
@luisarevalo21 my mistake, I realized the issues I brought up were only because I used Word; no problem copying & pasting into Notepad. Works as intended, good job!

**Pull request closed without merging**